enterprise/api/v1/views.py

```diff
@@ -1483,3 +1484,51 @@ def update(self, request, *args, **kwargs):
     )
     def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
+
+    @action(methods=['post'], detail=True, url_path='link-user')
+    def link_user(self, request, pk=None):
+        """
+        Post
+        Links user using enterprise_customer_key
+        /enterprise/api/enterprise_user_link/{enterprise_customer_key}/link-user
+
+        Given a enterprise_customer_key, link user to the appropriate enterprise.
+
+        If the key is not found, returns 400
+        If the ket is not valid, return 422
+        """
+        try:
+            enterprise_customer_key_match = models.EnterpriseCustomerInviteKey.objects.get(uuid=pk)
+
+        except models.EnterpriseCustomerInviteKey.DoesNotExist:
+            return Response({"error": "Could not find Enterprise Customer Invite Key"}, status=HTTP_400_BAD_REQUEST)
+
+        if not enterprise_customer_key_match.is_valid:
+            return Response(
+                {"detail": "Enterprise customer invite key is not valid"},
+                status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+
+        enterprise_customer = enterprise_customer_key_match.enterprise_customer
+
+        user, created = models.EnterpriseCustomerUser.objects.get_or_create(
+            user_id=request.user.id,
+            enterprise_customer=enterprise_customer,
+        )
+
+        if created:
+            user.invite_key = enterprise_customer_key_match
+            user.save()
+            track_enterprise_user_linked(
+                request.user.id,
+                pk,
+                enterprise_customer.uuid,
+            )
+        elif not user.active:
+            user.active = True
+            user.save()
+
+        return Response(
+            {"enterprise_customer_slug": enterprise_customer.slug},
```

Comment:
[suggestion] Might also be worth including the enterprise_customer.uuid in the response here?